### DISCUSSION TITLE
Shebang support

### DISCUSSION THREAD

**Discussion by gergelyk** [tiangolo/typer-cli#140] [Questions] [answered]
This could be useful if typer-cli supported using it in shebang: `#!/usr/bin/env typer`.

This would require `run` to be default command, so instead of `typer foo.py run bar` we would call it like `typer foo.py bar`.

Of course utils would need to be available in a different manner than now. For instance `typer --utils docs foo.py`.

**Comment by tiangolo** [comment, marked as answer]
`typer something.py run` is needed to be able to get autocompletion in the shell and all that stuff, which is pretty much the benefit of the `typer` command.

The shebang would be useful if you make the script executable and you want to call it directly. But in that case, you could use plain Python, and run typer at the end, e.g.:

* `main.py`:

```Python
#! /usr/bin/env python

import typer


def main(name: str):
    print(f"Hello {name}")


if __name__ == "__main__":
    typer.run(main)
```

And then:

```console
$ chmod +x main.py
```

And then you can use it as a script directly:

```console
$ ./main.py Grzegorz
Hello Grzegorz
```

---

Note: Typer-CLI is now part of Typer, you should simply install `pip install typer` and you would get the `typer` command. This repo and package, `typer-cli`, is no longer needed/useful. :rocket:
